build/azure-pipelines/win32/product-build-win32.yml

```
@@ -82,10 +82,22 @@ steps:
     condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'), ne(variables['NPM_REGISTRY'], 'none'))
     displayName: Setup NPM Authentication
 
+  - powershell: |
+      . build/azure-pipelines/win32/exec.ps1
+      $ErrorActionPreference = "Stop"
+      exec { git clone https://github.com/rzhao271/node-gyp.git .build/node-gyp } "Cloning rzhao271/node-gyp failed"
+      cd .build/node-gyp
+      exec { git checkout 102b347da0c92c29f9c67df22e864e70249cf086 } "Checking out 102b347 failed"
+      exec { npm install } "Building rzhao271/node-gyp failed"
+      cd ../..
+    displayName: Install custom node-gyp
+    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
```

```suggestion

  - powershell: |
      . build/azure-pipelines/win32/exec.ps1
      $ErrorActionPreference = "Stop"
      exec { mkdir -Force .build/node-gyp }
    displayName: Create custom node-gyp directory
    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
  
  - powershell: |
      . build/azure-pipelines/win32/exec.ps1
      $ErrorActionPreference = "Stop"
      # TODO: Should be replaced with upstream URL once https://github.com/nodejs/node-gyp/pull/2825
      # gets merged.
      exec { git clone https://github.com/rzhao271/node-gyp.git . } "Cloning rzhao271/node-gyp failed"
      exec { git checkout 102b347da0c92c29f9c67df22e864e70249cf086 } "Checking out 102b347 failed"
      exec { npm install } "Building rzhao271/node-gyp failed"
    displayName: Install custom node-gyp
    workingDirectory: .build/node-gyp
    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
```

Lets avoid relative `cd` operations, https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-powershell?view=azure-pipelines. Also added `TODO` comment for follow-ups.

---

The TODO comment is good. However, I suspect the "Install custom node-gyp" step will fail because `. build/azure-pipelines/win32/exec.ps1` doesn't work anymore once the working directory has changed. I'll adjust that line, but I also noticed that my attempt to verify the failure in CI didn't work because the step actually got skipped: https://dev.azure.com/monacotools/Monaco/_build/results?buildId=210496&view=results